governator-core/src/main/java/com/netflix/governator/internal/AbstractLifecycleFeature.java

```diff
@@ -17,39 +20,67 @@
  * 
  * @author elandau
  */
-abstract class AbstractLifecycleFeature implements LifecycleFeature {
+abstract class AbstractLifecycleFeature<C> implements LifecycleFeature {
+	enum IntrospectionFlag {
+		FIELD, METHOD, INTERFACE, SUPERCLASS
```

Comment:
INTERFACE would be optional if we want JSR-250-recommended handling of @PreDestroy and @PostConstruct.  SUPERCLASS would be optional if we only wanted to consider annotations in the top-level class. for instance we might process @Cacheable or @Bean only at the top-level to 'minimize astonishment' for the coder.  Practically I added SUPERCLASS to the enum for consistency's sake.  It would otherwise be the only one of four introspection control points with no 'feature switch', and the cost to add it was vanishingly small.